syntax = "proto3";

package pubsub;

service PubSubService {
  rpc SubscribeAndPublish (stream Message) returns (stream Message);
}

message Value {
  oneof value {
    string string_value = 1;
    int32 int_value = 2;
    double double_value = 3;
  }
}

message Message {
  string topic = 1;
  repeated Value values = 2;
}
